### PR TITLE
sys/od/od: fix -Werror=char-subscripts

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

`od_string` is not used in any application, put with current toolchain version in docker a warning appears

### Testing procedure

On master:

`BOARD=samr21-xpro DOCKER_ENVIRONMENT_CMDLINE='-e USEMODULE=od_string' BUILD_IN_DOCKER=1 make -C examples/hello-world/ `

```
In file included from /data/riotbuild/riotbase/sys/od/od.c:17:
/data/riotbuild/riotbase/sys/od/od.c: In function 'od_hex_dump':
/data/riotbuild/riotbase/sys/od/od.c:45:42: error: array subscript has type 'char' [-Werror=char-subscripts]
   45 |                 if(isprint(((char *)data)[str_pos+k])){
      |                            ~~~~~~~~~~~~~~^~~~~~~~~~~
cc1: all warnings being treated as errors
/data/riotbuild/riotbase/Makefile.base:108: recipe for target '/data/riotbuild/riotbase/examples/hello-world/bin/samr21-xpro/od/od.o' failed
```

Works with this PR.

```
READY
s
START
main(): This is RIOT! (Version: 2020.07-devel-1247-gce969-pr_fix_od_werror)
od_hex_dump(short_str, sizeof(short_str), OD_WIDTH_DEFAULT)
00000000  41  42  00                                                      AB.
od_hex_dump(long_str, sizeof(long_str), OD_WIDTH_DEFAULT)
00000000  FF  2C  61  FF  2E  62  63  64  65  66  67  68  69  6A  6B  6C  .,a..bcdefghijkl
00000010  6D  6E  6F  70  00                                              mnop.
od_hex_dump(long_str, sizeof(long_str), 4)
00000000  FF  2C  61  FF  .,a.
00000004  2E  62  63  64  .bcd
00000008  65  66  67  68  efgh
0000000C  69  6A  6B  6C  ijkl
00000010  6D  6E  6F  70  mnop
00000014  00              .
od_hex_dump(long_str, sizeof(long_str), 3)
00000000  FF  2C  61  .,a
00000003  FF  2E  62  ..b
00000006  63  64  65  cde
00000009  66  67  68  fgh
0000000C  69  6A  6B  ijk
0000000F  6C  6D  6E  lmn
00000012  6F  70  00  op.
od_hex_dump(long_str, sizeof(long_str), 8)
00000000  FF  2C  61  FF  2E  62  63  64  .,a..bcd
00000008  65  66  67  68  69  6A  6B  6C  efghijkl
00000010  6D  6E  6F  70  00              mnop.
All tests successful
```

### Issues/PRs references

Found while building #13824 in docker.

### CODE REVIEW

**Comment by aabadie** [comment]
Weird, why does it work on Murdock (which is supposed to use the same toolchain) ?

**Comment by fjmolinas** [comment]
> Weird, why does it work on Murdock (which is supposed to use the same toolchain) ?

The module is unused in master, its commented out in `tests/od`

**Comment by aabadie** [comment]
> The module is unused in master

So why not just add a test application ?

**Comment by fjmolinas** [comment]
> So maybe also add a test application ?

The test application already has a test for both, it just doesn't compile them both, should I split them?

**Comment by aabadie** [comment]
> it just doesn't compile them both, should I split them?

Ok, I see. Nevermind, this can be addressed in a follow-up.

**Comment by fjmolinas** [comment]
> Ok, I see. Nevermind, this can be addressed in a follow-up.

Maybe I should just comment in `od_string`?

**Comment by fjmolinas** [comment]
I commented in `od_string`, both tests overlap, so testing both at once make sense, no need for an extra application and the jobs that would come with :)

File: tests/od/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 include ../Makefile.tests_common
 
 USEMODULE += od
-# USEMODULE += od_string
+USEMODULE += od_string
```
Comment (aabadie) [inline tests/od/Makefile:4]: Then you can aso drop the block below and remove 02-run.py.

**Pull request merged**